extensions/objects/templates/literally-canvas-diagram-editor.component.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for literally canvas diagram editor.
+ */
+
+const LC = require('literallycanvas');
+require('literallycanvas/lib/css/literallycanvas.css');
+require('services/literally-canvas-helper.service.ts');
+
+angular.module('oppia').component('literallyCanvasDiagramEditor', {
+  template: require('./literally-canvas-diagram-editor.component.html'),
+  controller: [
+    'LiterallyCanvasHelperService', function(LiterallyCanvasHelperService) {
+      const ctrl = this;
+      var MAX_DIAGRAM_WIDTH = 491;
+      var MAX_DIAGRAM_HEIGHT = 551;
+      var DEFAULT_STROKE_WIDTH = 2;
+      var STROKE_WIDTH = [1, 2, 3, 5, 30];
```

Comment:
ALLOWED_STROKE_WIDTHS. The variable contains an array value so its name should be plural.

Reply:
Done.